pallets/rmrk-core/src/functions.rs

```diff
@@ -103,6 +107,22 @@ impl<T: Config> Property<KeyLimitOf<T>, ValueLimitOf<T>, T::AccountId, T::Collec
 		Self::deposit_event(Event::PropertyRemoved { collection_id, maybe_nft_id, key });
 		Ok(())
 	}
+
+	// Internal function to remove all of the properties for downstream
+	// `Origin::root()` calls.
+	fn do_remove_properties(
+		collection_id: T::CollectionId,
+		maybe_nft_id: Option<T::ItemId>,
+	) -> sp_runtime::DispatchResult {
+		let _ = Properties::<T>::clear_prefix(
+			(&collection_id, maybe_nft_id),
+			Properties::<T>::iter_prefix((collection_id, maybe_nft_id)).count() as u32,
```

Comment:
Suboptimal. When running `iter_prefix(...).count()`, it essentially go through the storage and read them all. So it will cause double expensive reads. The limit parameter is used to prevent unexpected large read, but by counting it, it doesn't help solve the problem.

I'd suggest to just drain the iterator, or just call `clear_prefix` with a limit specified from the caller.